evap/static/ts/src/csrf-utils.ts

```diff
@@ -28,6 +32,27 @@ $.ajaxSetup({
     },
 });
 
+function addCsrfToken(this: HTMLFormElement, ev: Event) {
+    const inputElement = document.createElement("input");
+    inputElement.setAttribute("type", "hidden");
+    inputElement.setAttribute("name", "csrfmiddlewaretoken");
+    inputElement.setAttribute("value", csrftoken);
+
+    const { submitter } = ev as SubmitEvent;
+    const action = (submitter && submitter.getAttribute("formaction")) || this.getAttribute("action");
```

Comment:
This can use the really fancy not-so-new (I was really surprised that not only ts, but also browsers have them already) operators:
```javascript
const action = submitter?.getAttribute("formaction") ?? form.getAttribute("action");
```